### PR TITLE
Improved SubFixer Capabilities

### PR DESCRIPTION
**Pull request by ssamadgh**
Now SubFixer Can accept batch files and folders, and check if they are
subtitle, fix their encoding, and if they are folder fix their inside
subtitle file if there is.
Also, now app checks if the subtitle file, is in true encode ignores it
and goes to next file checking if there.

### CODE REVIEW

File: SubFixer/Model/SFConverter.m [hunk 1]
```diff
@@ -10,27 +10,65 @@
...
+        if (![fileExtension isEqualToString:@"srt"]) {
+            BOOL isDir;
+            BOOL exists = [[NSFileManager defaultManager] fileExistsAtPath:path isDirectory:&isDir];
+            if (exists) {
+                /* file exists */
+                if (isDir) {
+                    /* file is a directory */
+                    NSArray * dir = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:path error:NULL];
+                    for (NSString *fileName in dir) {
+                        NSString *combinedPath = [NSString stringWithFormat:@"%@%@%@", path, @"/", fileName];
```
Comment (ahmadhashemi) [inline SubFixer/Model/SFConverter.m]: There is an issue here. When you find a folder and check it's content, you don't check if it's contents are srt files or not. So maybe you end up with an array full of multiple types of files.

**Comment by ahmadhashemi** [comment]
Everything seems fine. Thanks for the new features. But I have some concerns about checkIfEncodeIsUTF8String method. Have you checked it's functionality? Does it always return valid response? I mean, for example, when you pass it a valid Persian subtitle it return YES, and when you give it a broken one, it says NO?

**Comment by ssamadgh** [comment]
Hi about your first question, I've reviewed dragged files in two steps, first for discovering folders, and second for checking all founded files kind.
About your second question, yes, don't worry about that, it's working fine!

**Comment by ahmadhashemi** [comment]
I merged your commit into the main branch. I will try to do a clean up about new changes to make it more readable for other users. Thank you for your time and support.

**Pull request merged**